src/sherpa_ai/memory/belief.py

```diff
@@ -90,6 +90,63 @@ def get_internal_history(
 
         return context
 
+    def get_histories_without_result(
```

Comment:
We should make this method generic like `get_histories_excuding_types(self, token_counter, max_tokens, exclude_types)`, and the exclude types can be a list of event type to exclude when get the history